### PR TITLE
Add k-nearest centroid neighbor (kNCN) algorithm

### PR DESCRIPTION
**Pull request by T-Engel**
* rarefaction() has a new method "kNCN"
* Currently method "spat" is suppressed by `if (method == 'spat') method = 'kNCN'`

### CODE REVIEW

File: R/k-NCN_curve_functions.R [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Internal function for kNCN_average (). k-NCN algorithm starting with a specified focal sample
+
+centroid_accumulate<-function(x,focal_sample= 1, n=NULL, coords=NULL){
```
Comment (T-Engel) [inline R/k-NCN_curve_functions.R]: This function accumulates starting with a specified focal sample. It is the internal working horse for `kNCN_average()` that generates the average curves across all possible start samples.

File: R/k-NCN_curve_functions.R [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+#' @param x a mob_in object or a sites x species matrix
+#' @param n number of sites to include. 
+#' @param coords spatial coordinates of the samples. If x is a mob_in object, the fuction uses its "spat" table as coords.
+#' @param repetitions Number of times to repeat the procedure. Useful in situations where there are many ties in the distance matrix.
+#'
+#' @return a numeric vector of estimated species richness
+#' @export
+#'
+#' @examples
+kNCN_average<-function(x, n=NULL, coords=NULL, repetitions=1){
```
Comment (T-Engel) [inline R/k-NCN_curve_functions.R]: This is the new function that is called in `rarefaction()` for method = "kNCN"

**Comment by dmcglinn** [comment]
Hey @T-Engel thanks for submitting this PR let's hold off on merging this because for some reason a bunch of older commits are on this PR. I suppose this is probably because I let the `dev` branch get pretty out of date with `master`. I just want to make sure nothing is going to break here by doing this merge. So please be patient while I look this over.

**Comment by T-Engel** [comment]
Yes I also noticed that. I had already pulled from master and dev to incorporate the changes on both branches. So I thinking merging this will largely catch up dev with master. But it's probably best if you mergeaster into dev first.
________________________________________

**Comment by dmcglinn** [comment]
Hey @T-Engel I brought `dev` up-to-date with `master` to make the changes associated with your feature clearer. Unfortunately and not unexpectedly there are some merge conflicts that need to be resolved. I'm going to: 

1) close this PR
2) locally checkout your `kNCN` branch and clean up the conflicts
3) push the cleaned version up for a new PR

I just wanted to give you a heads up. Thanks again for this great contribution!

**Pull request closed without merging**